src/aws/mod.rs

```diff
@@ -201,25 +201,27 @@ pub async fn create_client_and_region<T: ClientBuilder>(
     };
 
     // Build the configuration first.
-    let mut config_builder = SdkConfig::builder()
-        .http_client(connector)
-        .sleep_impl(Arc::new(TokioSleep::new()))
-        .identity_cache(auth.credentials_cache().await?)
-        .credentials_provider(
+    let mut config_builder = SdkConfig::builder();
+
+    config_builder
+        .set_http_client(Some(SharedHttpClient::new(connector)))
+        .set_sleep_impl(Some(SharedAsyncSleep::new(Arc::new(TokioSleep::new()))))
+        .set_identity_cache(Some(auth.credentials_cache().await?))
+        .set_region(Some(region.clone()))
+        .set_retry_config(Some(retry_config.clone()))
```

Comment:
Why did these all become options? Somewhat confusing since the credentials provider doesn't appear to be one.

Reply:
Ah, good question, in hindsight I should have left a comment explaining this. This change was because I wanted to use `set_credentials_provider` to pass in an `Option` (since the credentials provider can be `None`). This  requires a `&mut self`, and modifies in place, rather than a `self` where it returns an updated `self`. I updated all of the calls for consistency but I think I could leave the existing calls and just use `set_credentials_provider` for the one where I need to pass an `Option` if preferred.

Reply:
The builder has `<option>` and `set_<option>` methods for all configuration options. The ones with `set_` take an `Option` (and `&mut self) where the ones without do not (and take a `self`).